lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtCleanUpStepImplTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.java;
+
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import com.diffplug.spotless.TestProvisioner;
+import com.diffplug.spotless.extra.EquoBasedStepBuilder;
+import com.diffplug.spotless.extra.eclipse.EquoResourceHarness;
+
+import org.assertj.core.util.Arrays;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.manipulation.CodeStyleConfiguration;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
+
+/** Eclipse JDT wrapper integration tests */
+public class EclipseJdtCleanUpStepImplTest  extends EquoResourceHarness {
+	private final static String INPUT = "package p; class C{}";
+	private final static String EXPECTED = "package p;\nclass C {\n}";
+	private static String SOURCE_FILE_PATH = "some .. / \\ ill&formatted/$path";
+	private static TestData TEST_DATA = null;
+
+	@BeforeAll
+	public static void initializeStatic() throws Exception {
+		TEST_DATA = TestData.getTestDataOnFileSystem();
+	}
+
+	private static EquoBasedStepBuilder createBuilder() {
+		return EclipseJdtCleanUpStep.createBuilder(TestProvisioner.mavenCentral());
+	}
+
+	public EclipseJdtCleanUpStepImplTest() {
+		super(createBuilder(), INPUT, EXPECTED);
+	}
+
+	@Test
+	public void emptyInput() throws Throwable {
+		cleanUpTest("", "", config -> {});
+	}
+
+	@Test
+	public void defaultConfiguration() throws Throwable {
+		for (String testFile : Arrays.array("Simple", "Statics", "Wildcards")) {
+			try {
+				cleanUpTest(testFile, config -> {});
+			} catch (AssertionFailedError e) {
+				throw new AssertionFailedError(testFile + " - " + e.getMessage(), e.getExpected(), e.getActual());
+			}
+		}
+	}
+
+	@Test
+	public void invalidConfiguration() throws Throwable {
+		//Smoke test, no exceptions expected
+		cleanUpTest("", "", config -> {
+			config.put("invalid.key", "some.value");
+		});
+		cleanUpTest("", "", config -> {
+			config.put(JavaCore.COMPILER_SOURCE, "-42");
```

Comment:
`JavaCore` is unknown here, while it is known in the `jdt.compile` module. How should I manage additional dependencies in test classes in lib-extra?

Reply:
@nedtwigg I would need a hint here.

Reply:
The build does not currently have a way to run tests against the Eclipse classpath. We would have to add something into this block

https://github.com/diffplug/spotless/blob/bc21f833c000b4095232bf211edacd8450a7ac2d/lib-extra/build.gradle#L47-L56


Along these lines

```
tasks.register("${needsP2}Test", Test) {
  // setup classpath somehow
}
...

into 'jdtCompileOnly', { ... }
into 'jdtTestImplementation', { ... }
```

There would be some copy pasting going on, if this got implemented then there'd be no copy-pasting necessary

- https://github.com/equodev/equo-ide/issues/109